model/task_lifecycle.go

```diff
@@ -796,11 +796,31 @@ func evalStepback(t *task.Task, caller, status string, deactivatePrevious bool)
 		if err != nil {
 			return errors.WithStack(err)
 		}
-		if shouldStepBack {
-			if err = doStepback(t); err != nil {
-				return errors.Wrap(err, "Error during step back")
+		if !shouldStepBack {
+			return nil
+		}
+
+		if t.IsPartOfSingleHostTaskGroup() {
+			// stepback earlier task group tasks as well
```

Comment:
I think that adding "because they need to run sequentially on the same host in order to work" to this comment might help people reading the code in the future, but up to you :)